test/integration/oidc/oidc_test.go

```diff
@@ -0,0 +1,435 @@
+			configureClientConfigForOIDC(apiServer.ClientConfig, expiredIDToken, stubRefreshToken, oidcServer.URL())
+			expiredClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+			configureOIDCServerToReturnExpiredRefreshTokenErrorOnTryingToUpdateIDToken(oidcServer)
+			_, err := expiredClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+			assert.Error(t, err)
+
+			tt.configureUpdatingTokenBehaviour(t, oidcServer)
+			idToken, stubRefreshToken := fetchOIDCCredentials(t, oidcServer.TokenURL(), caFilePath)
+			configureClientConfigForOIDC(apiServer.ClientConfig, idToken, stubRefreshToken, oidcServer.URL())
+			expectedOkClient := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+			_, err = expectedOkClient.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+
+			if tt.expectedErr != nil {
+				assert.ErrorAs(t, err, &tt.expectedErr)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestClientSecret(t *testing.T) {
+	var tests = []struct {
+		name                              string
+		configureOIDCServerTokenBehaviour func(t *testing.T, oidcServer *oidcTest.TestServer)
+		expectedErrMessage                string
+	}{
+		{
+			name: "client secret is not passed",
+			configureOIDCServerTokenBehaviour: func(t *testing.T, oidcServer *oidcTest.TestServer) {
+				oidcServer.TokenHandler().EXPECT().Token().Times(2).Return(oidcTest.Token{}, oidcTest.ErrClientSecretNotPassed)
+			},
+			expectedErrMessage: oidcTest.ErrClientSecretNotPassed.Error(),
+		},
+		{
+			name: "client secret is wrong",
+			configureOIDCServerTokenBehaviour: func(t *testing.T, oidcServer *oidcTest.TestServer) {
+				oidcServer.TokenHandler().EXPECT().Token().Times(2).Return(oidcTest.Token{}, oidcTest.ErrBadClientSecret)
+			},
+			expectedErrMessage: oidcTest.ErrBadClientSecret.Error(),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			oidcServer, apiServer := configureTestInfrastructure(t)
+
+			emptyIDToken, stubRefreshToken := "", defaultStubRefreshToken
+			configureClientConfigForOIDC(apiServer.ClientConfig, emptyIDToken, stubRefreshToken, oidcServer.URL())
+			client := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+
+			tt.configureOIDCServerTokenBehaviour(t, oidcServer)
+			_, err := client.CoreV1().Pods(defaultNamespace).List(context.Background(), metaV1.ListOptions{})
+			assert.Contains(t, err.Error(), tt.expectedErrMessage)
+		})
+	}
+}
+
+func TestWrongClientID(t *testing.T) {
+	oidcServer, apiServer := configureTestInfrastructure(t)
+
+	emptyIDToken, stubRefreshToken := "", defaultStubRefreshToken
+	configureClientConfigForOIDC(apiServer.ClientConfig, emptyIDToken, stubRefreshToken, oidcServer.URL())
+	client := kubernetes.NewForConfigOrDie(apiServer.ClientConfig)
+
+	oidcServer.TokenHandler().EXPECT().Token().Times(2).Return(oidcTest.Token{}, oidcTest.ErrBadClientID)
```

Comment:
you should check the client ID sent was in the specification-expected format according to the client auth type